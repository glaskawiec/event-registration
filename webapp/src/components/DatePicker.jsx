import React from 'react';
import PropTypes from 'prop-types';
import ValidationBadge from '../components/ValidationBadge';
import DateTimePicker from 'react-widgets/lib/DateTimePicker'

const DatePicker = (props) => (
  <>
    <DateTimePicker
      onChange={props.input.onChange}
      onFocus={props.input.onFocus}
      onBlur={props.input.onBlur}
      format="DD MMM YYYY"
      time={false}
      value={!props.input.value ? null : new Date(props.input.value)}
    />
    <ValidationBadge
      touched={props.meta.touched}
      error={props.meta.error} />
  </>
)

DatePicker.propTypes = {
  input: PropTypes.object.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired
};

export default DatePicker;