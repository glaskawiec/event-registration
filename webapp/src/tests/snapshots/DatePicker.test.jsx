import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from '../../components/DatePicker';
import Moment from 'moment'
import momentLocalizer from 'react-widgets-moment';

test('DatePicker snapshot', () => {
    Moment.locale('en');
    momentLocalizer();

    const props = {
        input: {
            onChange: () => ({})
        },
        meta: {
            touched: false,
            error: ''
        }
    };
    const tree = renderer
        .create(
            <DatePicker {...props} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});