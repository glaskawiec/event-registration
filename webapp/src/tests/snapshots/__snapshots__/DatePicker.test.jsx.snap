// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker snapshot 1`] = `
<div
  className="rw-datetime-picker rw-widget"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  tabIndex="-1"
>
  <div
    className="rw-widget-picker rw-widget-container"
  >
    <input
      aria-activedescendant={null}
      aria-expanded={false}
      aria-haspopup={true}
      aria-owns=""
      autoComplete="off"
      className="rw-widget-input rw-input"
      id="rw_1_input"
      onBlur={[Function]}
      onChange={[Function]}
      role="combobox"
      tabIndex="0"
      type="text"
      value=""
    />
    <span
      className="rw-select rw-select-bordered"
    >
      <button
        aria-label="Select date"
        className="rw-btn rw-btn-select"
        onClick={[Function]}
        tabIndex="-1"
        title="Select date"
        type="button"
      >
        <span
          aria-hidden="true"
          className="rw-i rw-i-calendar"
        />
      </button>
    </span>
  </div>
</div>
`;
